ucan.ipldsch

```diff
@@ -34,11 +34,47 @@ type Resource = String
 -- Must be all lower-case `/` delimeted with at least one path segment
 type Ability = String
 
+
+type Principal union {
+  -- Specification defines principals in terms of did:key's (that are multicodec
+  -- identifier for the public key type followed by the raw bytes of the key)
+  -- we represent those as raw public key bytes prefixed with public key multiformat
+  -- code.
+  | Ed25519    "0xed"
+  | RSA        "0x1205"
+  | P256       "0x1200"
+  | P384       "0x1201"
+  | P521       "0x1202"
+  
```

Comment:
```suggestion
  | X25519 "0xec"
  | BLS12381G1 "0xea"
```

I THINK it's BLS-12381 _G1_ not _G2_ in `did:key`, but I need to check

Reply:
Yeah I'm like 99% sire that G1 is for public keys, and G2 is for signatures. Pairing-based cyrpto, I tell ya.